examples/echo_server/include/lwip/lwipopts.h

```diff
@@ -37,13 +38,27 @@
 #define CHECKSUM_CHECK_ICMP             0
 #define CHECKSUM_CHECK_ICMP6            0
 
+#ifdef CONFIG_PLAT_IMX8MM_EVK
```

Comment:
I don't think we should have platform dependent code inside `lwipopts.h`. I imagine that the checksum options are different for the Odroid from debugging the driver?

Reply:
The odroidc4 requires the generation of checksums by software whereas the IMX8 doesn't. This is not a relic of debugging the driver, if we don't configure lwip to generate checksums no packets will successfully be sent. This lwipopts is echo server specific and the echo server explicitly only supports these two boards. So I don't see an issue with putting the ifdef in lwipopts.

Reply:
should use a feature-test macro, something like 'ifdef HWCHECKSUM' --- then define that for imx8 but not for odroid.